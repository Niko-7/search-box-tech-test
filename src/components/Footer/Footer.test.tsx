/*eslint-disable */
import { Footer } from ".";
import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
const { axe, toHaveNoViolations } = require("jest-axe");
expect.extend(toHaveNoViolations);

afterEach(() => {
  cleanup();
});

describe("Footer component", () => {
  it("should render Footer component", () => {
    render(<Footer />);
    expect(<Footer />).toBeInTheDocument;
  });
});

describe("accesibility tests", () => {
  it("should have a footer label", () => {
    render(<Footer />);
    const footerLabel = screen.getByLabelText("Information Footer");
    expect(footerLabel).toBeInTheDocument;
  });

  it("should have no accesibility errors on Footer component", async () => {
    const { container } = render(<Footer />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});

describe("Snapshot test", () => {
  it("should match the snapshot of the Footer component", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
