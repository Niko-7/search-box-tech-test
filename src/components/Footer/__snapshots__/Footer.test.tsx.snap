// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should match the snapshot of the Footer component 1`] = `
<footer
  aria-label="Information Footer"
>
  <div
    className="information-footer"
  >
    <a
      data-alt="Linked In Page"
      href="https://www.linkedin.com/in/nfloros/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <p>
        Nikolaos Floros
      </p>
    </a>
    <p
      className="user-information"
    >
      Front-End Technical Test
    </p>
    <br />
    <a
      data-alt="Github Repo Source Code Page"
      href="https://github.com/Niko-7/search-box-tech-test"
      rel="noopener noreferrer"
      target="_blank"
    >
      <p>
        View Source Code
      </p>
    </a>
  </div>
</footer>
`;
